File: app/src/main/java/com/caramelheaven/lennach/models/model/board/Board.java
package com.caramelheaven.lennach.models.model.board;

import java.util.List;
import java.util.Objects;

public class Board {
    private String board;
    private String boardName;
    private Integer boardSpeed;
    private Integer bumpLimit;
    private Integer maxComment;
    private Integer maxFileSize;
    private List<Integer> pages;
    private Integer currentPage;
    private List<Usenet> usenetList;

    @Override
    public String toString() {
        return "Board{" +
                "board='" + board + '\'' +
                ", boardName='" + boardName + '\'' +
                ", boardSpeed=" + boardSpeed +
                ", bumpLimit=" + bumpLimit +
                ", maxComment=" + maxComment +
                ", maxFileSize=" + maxFileSize +
                ", pages=" + pages +
                ", currentPage=" + currentPage +
                ", usenetList=" + usenetList +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Board board1 = (Board) o;
        return Objects.equals(board, board1.board) &&
                Objects.equals(boardName, board1.boardName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(board, boardName);
    }

    public Integer getCurrentPage() {
        return currentPage;
    }

    public void setCurrentPage(Integer currentPage) {
        this.currentPage = currentPage;
    }

    public List<Integer> getPages() {
        return pages;
    }

    public void setPages(List<Integer> pages) {
        this.pages = pages;
    }

    public String getBoard() {
        return board;
    }

    public String getBoardName() {
        return boardName;
    }

    public Integer getBoardSpeed() {
        return boardSpeed;
    }

    public Integer getBumpLimit() {
        return bumpLimit;
    }

    public Integer getMaxComment() {
        return maxComment;
    }

    public Integer getMaxFileSize() {
        return maxFileSize;
    }

    public List<Usenet> getUsenetList() {
        return usenetList;
    }

    public void setBoard(String board) {
        this.board = board;
    }

    public void setBoardName(String boardName) {
        this.boardName = boardName;
    }

    public void setBoardSpeed(Integer boardSpeed) {
        this.boardSpeed = boardSpeed;
    }

    public void setBumpLimit(Integer bumpLimit) {
        this.bumpLimit = bumpLimit;
    }

    public void setMaxComment(Integer maxComment) {
        this.maxComment = maxComment;
    }

    public void setMaxFileSize(Integer maxFileSize) {
        this.maxFileSize = maxFileSize;
    }

    public void setUsenetList(List<Usenet> usenetList) {
        this.usenetList = usenetList;
    }
}

File: app/src/main/java/com/caramelheaven/lennach/models/model/Board.kt
package com.caramelheaven.lennach.models.model

/**
 * Created by CaramelHeaven on 16:19, 03/02/2019.
 * Change to data class
 */
class Board {
    var board: String? = null
    var boardName: String? = null
    var boardSpeed: Int? = null
    var bumpLimit: Int? = null
    var maxComment: Int? = null
    var maxFileSize: Int? = null
    var pages: List<Int>? = null
    var currentPage: Int? = null
    var usenetList: List<Usenet>? = null

    override fun toString(): String {
        return "Board(board=$board, boardName=$boardName, boardSpeed=$boardSpeed, bumpLimit=$bumpLimit, maxComment=$maxComment, maxFileSize=$maxFileSize, pages=$pages, currentPage=$currentPage, usenetList=$usenetList)"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Board

        if (board != other.board) return false
        if (boardName != other.boardName) return false
        if (boardSpeed != other.boardSpeed) return false
        if (bumpLimit != other.bumpLimit) return false
        if (maxComment != other.maxComment) return false
        if (maxFileSize != other.maxFileSize) return false
        if (pages != other.pages) return false
        if (currentPage != other.currentPage) return false
        if (usenetList != other.usenetList) return false

        return true
    }

    override fun hashCode(): Int {
        var result = board?.hashCode() ?: 0
        result = 31 * result + (boardName?.hashCode() ?: 0)
        result = 31 * result + (boardSpeed ?: 0)
        result = 31 * result + (bumpLimit ?: 0)
        result = 31 * result + (maxComment ?: 0)
        result = 31 * result + (maxFileSize ?: 0)
        result = 31 * result + (pages?.hashCode() ?: 0)
        result = 31 * result + (currentPage ?: 0)
        result = 31 * result + (usenetList?.hashCode() ?: 0)
        return result
    }


}
